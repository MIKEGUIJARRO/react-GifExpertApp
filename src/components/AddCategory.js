import React, { useState } from "react";
import PropTypes from "prop-types";
//Aqui recibimos las (props) de AddCategory 
export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState("");

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 2) {
            setInputValue("");
            setCategories((items) => {
                return [inputValue, ...items];
            });
        }
    };

    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input type="text" value={inputValue} onChange={handleInputChange} />
        </form>
    );
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
}
