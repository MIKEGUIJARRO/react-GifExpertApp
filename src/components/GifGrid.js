import React, { Fragment } from "react";
import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGifs(category);

    return (
        <Fragment>
            <h3>{category}</h3>
            {loading && <p className="animate__animated animate__flash">Loading ...</p>}
            <div className="card-grid">
                {
                    images.map((image) => {
                        return <GifGridItem key={image.id} {...image} />
                    })
                }
            </div>
        </Fragment>
    );
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}