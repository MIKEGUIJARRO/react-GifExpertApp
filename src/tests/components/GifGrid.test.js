import { shallow } from "enzyme";
import React from "react";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("<GifGrid /> component test", () => {
    const category = "categoria1";
    test("Properly shows <GifGrid />", () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });

        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();
    });

    test("Should show items when the images are loaded with useFetchGifs hook", () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://google.com',
            title: 'ABC Title'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });

        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("p").exists()).toBe(false);
        expect(wrapper.find("GifGridItem").length).toBe(gifs.length)

    });

});