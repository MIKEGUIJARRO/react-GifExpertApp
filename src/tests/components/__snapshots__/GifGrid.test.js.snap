// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGrid /> component test Properly shows <GifGrid /> 1`] = `
<Fragment>
  <h3>
    categoria1
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading ...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`<GifGrid /> component test Should show items when the images are loaded with useFetchGifs hook 1`] = `
<Fragment>
  <h3>
    categoria1
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="ABC Title"
      url="https://google.com"
    />
  </div>
</Fragment>
`;
